homeassistant/components/telegram_bot/webhooks.py

```diff
@@ -35,44 +42,58 @@ async def async_setup_platform(hass, bot, config):
 
     if not pushbot.webhook_url.startswith("https"):
         _LOGGER.error("Invalid telegram webhook %s must be https", pushbot.webhook_url)
-        return False
+        return None
```

Comment:
It's not good that we can't distinguish an error during bot app setup, and no bot app (as returned by the broadcast platform). We should make sure that all platforms conform to the same interface and that the return value from the interface means that same thing.

Maybe we should raise and exception on error instead of returning `None` here?

If there's an error here we want to fail the config entry setup, either by returning `False` from `async_setup_entry` or by raising `ConfigEntryNotReady` inside `async_setup_entry` (to retry the config entry setup, for retryable errors).